src/Infrastructure/GlobalSequentialIdGenerator.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\PersistentPageIdentifiers\Infrastructure;
+
+class GlobalSequentialIdGenerator implements IdGenerator {
+
+	private static int $counter = 0;
```

Comment:
:scream:

Reply:
It was necessary to keep track across all tests, otherwise each test starts with 0, which leads to DB constraint violation.